File: Data Structures & Algorithms/first-missing-positive/submission-1.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int len=nums.length;
        int res=1;
        for(int i=0;i<len;i++){
            if(nums[i]<0){
                nums[i]=0;
            }
        }
        for(int i=0;i<len;i++){
            int val=Math.abs(nums[i]);
            int index=val-1;
            if(index>=0&&index<len)
            {   
                if(nums[index]>0)nums[index]=nums[index]*-1;
                else if(nums[index]==0) nums[index]=(len+1)*-1;
            }
        }

        for(int i=0;i<len;i++){
            if(nums[i]>=0)
            {
                //res=i+1;
                return i+1;
                //break;
            }
        }
    return len+1;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res=[]
        def dfs(i,curr,total):
            if target==total:
                res.append(curr.copy())
                return
            if i==len(nums) or total>target:
                return
            curr.append(nums[i])
            dfs(i,curr,total+nums[i])
            curr.pop()
            dfs(i+1,curr,total)

        dfs(0,[],0)
        return res
        #TC:O(2^t/m)
        #SC:O(t/m)


        